e2e/testenv/kubernetes/kind/kind.go

```diff
@@ -0,0 +1,246 @@
+	if err := common.DeleteTestDeployment(e.Context(ctx)); err != nil {
+		return fmt.Errorf("failed to delete test deployment: %w", err)
+	}
+	if err := e.ChartHelper.DeleteHelmChart(); err != nil {
+		return fmt.Errorf("failed to delete VMclarity helm chart: %w", err)
+	}
+
+	return nil
+}
+
+// nolint:wrapcheck
+func (e *KindEnv) SetUp(ctx context.Context) error {
+	providerOpts, err := cluster.DetectNodeProvider()
+	if err != nil {
+		return fmt.Errorf("failed to detect provider: %w", err)
+	}
+	provider := cluster.NewProvider(providerOpts)
+	if err := provider.Create(e.name,
+		cluster.CreateWithConfigFile(e.kindConfigPath),
+		cluster.CreateWithWaitForReady(KindClusterCreationTimeout),
+	); err != nil {
+		return fmt.Errorf("failed to create kind cluster: %w", err)
+	}
+	e.provider = provider
+	if err := provider.ExportKubeConfig(e.name, e.kubeConfigPath, false); err != nil {
+		return fmt.Errorf("failed to get kubeconfig for kind cluster: %w", err)
+	}
+
+	return e.loadContainerImagesToCluster(ctx)
+}
+
+// nolint:wrapcheck
+func (e *KindEnv) TearDown(_ context.Context) error {
+	return e.provider.Delete(e.name, e.kubeConfigPath)
+}
+
+// nolint:wrapcheck
+func (e *KindEnv) ServicesReady(ctx context.Context) (bool, error) {
+	// NOTE(pebalogh) checking services isn't necessary because of the `helm install --wait`
+	return true, nil
+}
+
+// nolint:wrapcheck
+func (e *KindEnv) ServiceLogs(ctx context.Context, services []string, startTime time.Time, stdout, _ io.Writer) error {
+	ctx = e.Context(ctx)
+	for _, podName := range services {
+		pod, err := utils.GetVMClarityPodByName(ctx, podName, common.VMClarityNamespace)
+		if err != nil {
+			return fmt.Errorf("failed to get pod: %w", err)
+		}
+		logBytes, err := utils.GetPodLogs(ctx, pod, startTime)
+		if err != nil {
+			return fmt.Errorf("failed to get log for pod %s: %w", podName, err)
+		}
+		_, err = stdout.Write(logBytes)
+		if err != nil {
+			return fmt.Errorf("failed to write logs to stdout: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// nolint:wrapcheck
+func (e *KindEnv) Services(ctx context.Context) ([]string, error) {
+	podList, err := utils.ListVMClarityPods(e.Context(ctx), common.VMClarityNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list VMClarity pods: %w", err)
+	}
+
+	podNames := make([]string, 0)
+	for _, pod := range podList.Items {
+		podNames = append(podNames, pod.GetName())
+	}
+
+	return podNames, nil
+}
+
+// nolint:wrapcheck
+func (e *KindEnv) VMClarityAPIURL() (*url.URL, error) {
+	return &url.URL{
+		Scheme: "http",
+		Host:   fmt.Sprintf("localhost:%s/api", KindAPIServerPort),
+	}, nil
+}
+
+// nolint:wrapcheck
+func (e *KindEnv) Context(ctx context.Context) context.Context {
+	return context.WithValue(ctx, envtypes.KubernetesContextKey, e.k8sClientSet)
+}
+
+// nolint:wrapcheck
+func (e *KindEnv) loadContainerImagesToCluster(ctx context.Context) error {
+	nodeList, err := e.provider.ListNodes(e.name)
```

Comment:
if you use "kind load docker-image" command, then you don't need to load the image to each node.

Reply:
See what we did in Kubeclarity: https://github.com/openclarity/kubeclarity/blob/main/e2e/main_test.go#L52
I think it can replace all the logic here of loading an image to cluster.

Reply:
The e2e-framework is using `exec.command` for doing the image load. I wanted to avoid to use `exec.command` in the case of kind cluster start, stop, setup, etc...

Reply:
We need to load images to all nodes if we are using the kind package instead of `exec.command`